core/templates/cowdata.h

```diff
@@ -269,22 +269,28 @@ void CowData<T>::_unref() {
 
 	SafeNumeric<USize> *refc = _get_refcount();
 	if (refc->decrement() > 0) {
-		return; // still in use
+		// Data are still in use elsewhere.
+		_ptr = nullptr;
+		return;
 	}
-	// clean up
+	// Clean up.
+	// First, invalidate our own reference.
```

Comment:
I would like to see this comment elaborate a bit on the `refcount == 0` case. I think the root of the problem is the observability of _ptr while refcount == 0. I think describing the situation in which that can happen is important, but I think it misses the root of the issue.

Reply:
You're right, the problem can be better described more generally through exposing half-destructed data. I updated the comment, let me know if you think it reads better now!

Reply:
Equally perfect. Thanks